mindsdb_native/libs/controllers/transaction.py

```diff
@@ -359,7 +359,12 @@ def run(self):
                                 diff = sample[1, idx] - sample[0, idx]
                                 if diff <= tolerance:
                                     output_data[f'{predicted_col}_confidence'][sample_idx] = significance
-                                    output_data[f'{predicted_col}_confidence_range'][sample_idx] = list(sample[:, idx])
+                                    conf_range = list(sample[:, idx])
+
+                                    # for positive numerical domains
+                                    if self.lmd['stats_v2'][predicted_col].get('positive_domain', False):
+                                        conf_range[0] = max(0, conf_range[0])
```

Comment:
what if `conf_range` at indices greater than `0` has negative values?

Reply:
Good catch, that _could_ happen

Reply:
Scratch that, it cannot happen. We get a guarantee from Lightwood that predictions are `>= 0` for positive domains, and the sigma from the conformal predictor can never be negative. Still, I'll add a test for this.